h2o-py/h2o/explanation/_explain.py

```diff
@@ -1305,35 +1308,59 @@ def ice_plot(
             factor_map = _factor_mapper(NumpyFrame(frame[column]).from_factor_to_num(column))
 
         deciles = [int(round(frame.nrow * dec / 10)) for dec in range(11)]
+        deciles[10] = frame.nrow - 1
         colors = plt.get_cmap(colormap, 11)(list(range(11)))
         plt.figure(figsize=figsize)
         for i, index in enumerate(deciles):
+            percentile_string = "{}th Percentile".format(i * 10)
             tmp = NumpyFrame(
                 model.partial_plot(
                     frame,
                     cols=[column],
                     plot=False,
                     row_index=index,
                     targets=target,
-                    nbins=20 if not is_factor else 1 + frame[column].nlevels()[0]
+                    nbins=100 if not is_factor else 1 + frame[column].nlevels()[0],
+                    include_na=True
                 )[0]
             )
+            user_splits = dict()
+            orig_value = frame.as_data_frame()[column][index]
             encoded_col = tmp.columns[0]
+            if _isnan(orig_value):
+                id = np.where(tmp[encoded_col] == tmp.from_factor_to_num(encoded_col)['.missing(NA)'])[0][0] if is_factor else np.where(np.isnan(tmp[encoded_col]))[0][0]
+                orig_null_value = tmp.from_num_to_factor(encoded_col)[tmp[encoded_col][id]] if is_factor else tmp[encoded_col][id]
```

Comment:
nit/style: Use complete if/else statements instead of conditional expressions, as the cases here are complicated and hard to parse. I.e. use `x = A if y else B` only for simple cases. This is same for previous line

Reply:
[77c3d0a](https://github.com/h2oai/h2o-3/pull/6020/commits/77c3d0acca2ed68f6c134d67015734bc7c5af0f3)